app/src/main/java/fragments/TempFeedFragment.java

```diff
@@ -2,29 +2,112 @@
 import com.example.surfstop.R;
+import com.parse.FindCallback;
+import com.parse.ParseException;
+import com.parse.ParseQuery;
+
+import java.util.ArrayList;
+import java.util.List;
+
+import models.Post;
+import models.ShortPost;
 
 public class TempFeedFragment extends Fragment {
 
+    public static final String TAG = "TempFeedActivity";
+
+    private RecyclerView rvTempFeed;
+    protected List<Post> allPosts;
+    protected PostAdapter adapter;
+    SwipeRefreshLayout swipeContainer;
+
     public TempFeedFragment() {
         // Required empty public constructor
     }
 
-    @Override
-    public void onCreate(Bundle savedInstanceState) {
-        super.onCreate(savedInstanceState);
-    }
-
     @Override
     public View onCreateView(LayoutInflater inflater, ViewGroup container,
                              Bundle savedInstanceState) {
         // Inflate the layout for this fragment
         return inflater.inflate(R.layout.fragment_temp_feed, container, false);
     }
+
+    @Override
+    public void onViewCreated(@NonNull View view, @Nullable Bundle savedInstanceState) {
+        super.onViewCreated(view, savedInstanceState);
+        rvTempFeed = view.findViewById(R.id.rvTempFeed);
+
+        // initialize the array that will hold posts and create a PostsAdapter
+        allPosts = new ArrayList<>();
+        adapter = new PostAdapter(getContext(), allPosts);
+        swipeContainer = (SwipeRefreshLayout) view.findViewById(R.id.swipeContainer);
+
+        // set the adapter on the recycler view
+        rvTempFeed.setAdapter(adapter);
+        // set the layout manager on the recycler view
+        rvTempFeed.setLayoutManager(new LinearLayoutManager(getContext()));
+
+        Log.i(TAG, "before queryPosts");
+        queryPosts();
+        Log.i(TAG, "after queryPosts");
+
+        // query more posts
+        swipeContainer.setOnRefreshListener(new SwipeRefreshLayout.OnRefreshListener() {
+            @Override
+            public void onRefresh() {
+                // Your code to refresh the list here.
+                // Make sure you call swipeContainer.setRefreshing(false)
+                // once the network request has completed successfully.
+                adapter.clear();
+                queryPosts();
+                swipeContainer.setRefreshing(false);
+            }
+        });
+        // Configure the refreshing colors
+        swipeContainer.setColorSchemeResources(android.R.color.holo_blue_bright,
+                android.R.color.holo_green_light,
+                android.R.color.holo_orange_light,
+                android.R.color.holo_red_light);
+    }
+
+    protected void queryPosts() {
+        ParseQuery<ShortPost> query = ParseQuery.getQuery(ShortPost.class);
+        query.include(ShortPost.KEY_USER);
+        // Set number of items queried
+        query.setLimit(20);
+        // Order posts by creation date (newest first)
+        query.addDescendingOrder("createdAt");
```

Comment:
@bchang7 Yes, it is just from the parse api. There is also an updatedAt key, should I use that instead? If so, why is that preferable?

Reply:
i would imagine if a post is modified, maybe new information or new pictures, it may want to be refreshed to the top of the feed again? up to you though on what you think should be the right behavior.